pudl/load.py

```diff
@@ -28,24 +29,35 @@ def _csv_dump_load(df, table_name, engine, csvdir='', keep_csv=True):
             object, and to name the CSV file.
         engine (sqlalchemy.engine): SQLAlchemy database engine, which will be
             used to pull the CSV output into the database.
-        csvdir (str): Path to the directory into which the CSV files should be
-            output (and saved, if they are being kept).
+        csvdir (str): Path to the directory into which the CSV file should be
+            saved, if it's being kept.
         keep_csv (bool): True if the CSV output should be saved after the data
             has been loaded into the database. False if they should be deleted.
-
+            NOTE: If multiple COPYs are done for the same table_name, only
+            the last will be retained by keep_csv, which may be unsatisfying.
     Returns: Nothing.
     """
     import postgres_copy
-    import os
+    import io
 
-    csvfile = os.path.join(csvdir, table_name + '.csv')
-    df.to_csv(csvfile, index=False)
     tbl = pudl.models.entities.PUDLBase.metadata.tables[table_name]
-    with open(csvfile, 'r', encoding='utf8') as f:
+    # max_size is in bytes; spill to disk if >2GB
```

Comment:
Is this true, and set somewhere else? For the FERC/EIA tables, it looks like you've just swapped the in-memory file object for the on-disk CSV (which is great) and the 2GB limit and all other changes are implemented only in the CEMS loading process, right?

Now that there are substantially different load processes for CEMS vs. the other data sources, does it make sense to create a load subpackage with generic and data source specific modules in it?

Reply:
Sure, if that makes more sense. My thought was to write `BulkCopy` to be general enough to work with other data that might benefit from batching, but I don't know the EIA or FERC processing nearly as well and didn't want to change too many things.